src/libraries/System.Console/src/System/ConsolePal.iOS.cs

```diff
@@ -1,23 +1,89 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics;
 using System.IO;
+using System.Runtime.InteropServices;
 using System.Text;
 
 namespace System
 {
     internal sealed class NSLogStream : ConsoleStream
     {
-        public NSLogStream() : base(FileAccess.Write) {}
+        private readonly StringBuilder _buffer = new StringBuilder();
+        private Decoder _decoder;
+
+        public NSLogStream() : base(FileAccess.Write)
+        {
+            _decoder = Encoding.Unicode.GetDecoder();
+        }
 
         public override int Read(Span<byte> buffer) => throw Error.GetReadNotSupported();
 
         public override unsafe void Write(ReadOnlySpan<byte> buffer)
         {
-            fixed (byte* ptr = buffer)
+            int charCount = _decoder.GetCharCount(buffer, false);
+            Span<char> charSpan = new char[charCount];
```

Comment:
Can this array be pooled?

Reply:
Could you please clarify a bit with an example?

Reply:
Every time this method is called, it is allocating a new array just to use as a scratch pad, and then thrown away. If we pool the array instead, we don't need to allocate a new array every time.

Here's an example:

https://github.com/dotnet/runtime/blob/60af14837e78d69d5469c2c0c7632cfca761aa09/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L66-L78

Reply:
Thanks!